toolchain/parse/context.h

```diff
@@ -281,6 +281,13 @@ class Context {
   // Propagates an error up the state stack, to the parent state.
   auto ReturnErrorOnState() -> void { state_stack_.back().has_error = true; }
 
+  // Adds a node expecting a decl semi. Includes error recovery when the token
+  // is not a semi. `is_def_allowed` is only selecting the diagnostic if not a
+  // semi; a definition must be handled separately.
```

Comment:
Could you explain the arguments more explicitly:
- `decl_kind` I think is only used in error message
- `node_kind` is the kind of the node being added

Reply:
> Could you explain the arguments more explicitly?
> 
> * `decl_kind` I think is only used in error message

Done.

> * `node_kind` is the kind of the node being added

Can you please elaborate what you're wanting here? There's only one mention of a node in the comment, so I'm missing possible other meanings: "`node_kind` is the kind of the node" sounds redundant to me, but lacking an understanding of the ambiguity I'm not sure what else to write. Note this is also similar to other `AddNode` and `AddLeafNode` functions.